sunpy/net/attr.py

```diff
@@ -23,16 +16,76 @@
 Please note that & is evaluated first, so A & B | C is equivalent to
 (A & B) | C.
 """
+import keyword
+import warnings
+from collections import defaultdict, namedtuple
 
-from __future__ import absolute_import
+from astropy.utils.misc import isiterable
 
 from sunpy.util.multimethod import MultiMethod
 from sunpy.extern.six import iteritems
 
-# XXX: Maybe allow other normal forms.
+_ATTR_TUPLE = namedtuple("attr", "name name_long desc")
+
+__all__ = ['Attr', 'DummyAttr', 'SimpleAttr', 'AttrAnd',
+           'AttrOr', 'ValueAttr', 'and_', 'or_']
+
+
+def make_tuple():
+    return _ATTR_TUPLE([], [], [])
+
 
+class AttrMeta(type):
+    """
+    We want to enable automatic discovery via tab completion of subclasses of Attrs.
+    To do this we have to create a metaclass that redefines the methods that Python uses to normally do this.
```

Comment:
this. *So that they work on the class objects themselves*. 

(To differentiate between implementing them on the class.